### ISSUE TITLE
CA1508 (Avoid Dead Conditional Code) using statement continues to give false positive 

### ISSUE THREAD

**Issue by abatishchev** [dotnet/roslyn-analyzers#3064]
The continuation of #2250 as the latest version still has this (or similar) issue.

Would be great if you could prioritize the resolution as an internal partner's ask, Identity Platform Engineering would be grateful.

#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.7 (Latest)

#### Diagnostic ID

CA1508 (no article at https://docs.microsoft.com/en-us/visualstudio/code-quality/)

#### Repro steps

```c#
using (var fabricClient = new FabricClient())
{
}
```

#### Expected behavior

No error

#### Actual behavior

>'fabricClient = new FabricClient()' is never 'null'

More examples

![image](https://user-images.githubusercontent.com/351644/69371638-17d7e600-0c55-11ea-8b62-721ef03ac9a7.png)

**Comment by sharwell** [comment]
Duplicate of #3063

**Comment by abatishchev** [comment]
Sorry, bogus network connection created 2 issues instead of 1.

**Comment by sharwell** [comment]
Yep it's no problem at all 😄
